CONFIGURE.md

```diff
@@ -54,39 +54,35 @@ $ bundle exec rails console
 
 ## OAuth Consumer Keys
 
-There are two built-in applications which communicate via the API, and therefore need OAuth consumer keys configured. These are:
+There are two built-in applications which communicate via the API, and therefore need to be registered as OAuth 2 applications. These are:
 
 * iD
 * The website itself (for the Notes functionality)
 
-To use the iD editor you need to register it as an OAuth 1 application.
+For iD, do the following:
 
-Do the following:
-* Log into your Rails Port instance - e.g. http://localhost:3000
-* Click on your user name to go to your user page
-* Click on "my settings" on the user page
-* Click on "OAuth 1 settings" on the My settings page
-* Click on 'Register your application'.
+* Go to "[OAuth 2 applications](http://localhost:3000/oauth2/applications)" on the My settings page.
+* Click on "Register new application".
 * Unless you have set up alternatives, use Name: "Local iD" and Main Application URL: "http://localhost:3000"
 * Check boxes for the following Permissions
-  * 'read their user preferences'
-  * 'modify the map'
-  * 'read their private GPS traces'
-  * 'modify notes'
-* Everything else can be left with the default blank values.
-* Click the "Register" button
-* On the next page, copy the "consumer key"
+  * 'Read user preferences'
+  * 'Modify user preferences'
+  * 'Modify the map'
+  * 'Read private GPS traces'
+  * 'Upload GPS traces'
+  * 'Modify notes'
```

Comment:
In 2.21.0, iD requests the `write_prefs` scope (see https://github.com/openstreetmap/iD/blob/c79dfc601f794bab9d1fb7a27f00fd50d0d282c9/modules/services/osm.js#L23). AFAICS it is not (yet) used anywhere obvious. (//cc @bhousel: Did I overlooking something here?) I was however planning to make use of it in the not so far future to preserve some settings and/or editor states. That's why I choose to just keep it in.

`write_gpx` is currently not used by iD (and also not requested during authentication), but I thought about adding a functionality to upload traces from within iD at some point in the future.

I assumed that for a typical user setting up this repo, it would be less trouble to allow all potentially/eventually used permissions right now, as this would prevent hard to debug issues when the scopes would be used by iD at some point in the future.

Let me know if you prefer to handle it differently. I can also prepare a patch release which doesn't request `write_prefs`.

Reply:
> AFAICS it is not (yet) used anywhere obvious. (//cc @bhousel: Did I overlooking something here?) I was however planning to make use of it in the not so far future to preserve some settings and/or editor states. That's why I choose to just keep it in.

Yes `write_prefs` is not used currently, but it's been something we've considering for a long time and might use eventually: https://github.com/openstreetmap/iD/issues/3002